src/torchsim/models/mprage.py

```diff
@@ -140,16 +140,16 @@ def _engine(
         states = epg.adiabatic_inversion(states, inv_efficiency)
         states = epg.longitudinal_relaxation(states, E1inv, rE1inv)
         states = epg.spoil(states)
-
+        signal = []
         # Scan loop
         for p in range(nshots_bef):
 
             # Apply RF pulse
             states = epg.rf_pulse(states, RF)
-
             # Evolve
             states = epg.longitudinal_relaxation(states, E1, rE1)
```

Comment:
Without multiple repetitions, everything that happens after ADC should be irrelevant.